src/app/app-routing.ts

```diff
@@ -51,7 +52,8 @@ const requireLogin = (activatedRoute: ActivatedRouteSnapshot) => {
   // on successful login the login page redirects back to My Account by default.
   // If we need to redirect to any other pages in future, we can take an ActivatedRouteSnapshot param here
   // and pass it to the login page as an 'r' query param.
-  const redirectPath = activatedRoute?.routeConfig?.path;
+
+  const redirectPath = routerStateSnapshot.url
```

Comment:
activate route path didn't include the actual params, e.g the campaign ID in this case. Hasn't been an issue up to now because none of the routes that have used this requireLogin guard have had path parameters.